src/dist/replication/lib/replica_stub.cpp

```diff
@@ -889,6 +889,34 @@ void replica_stub::on_query_replica_info(const query_replica_info_request &req,
     resp.err = ERR_OK;
 }
 
+void replica_stub::on_query_disk_info(const query_disk_info_request &req,
+                                      /*out*/ query_disk_info_response &resp)
+{
+    for (const auto &dir_node : _fs_manager._dir_nodes) {
+        disk_info info;
+        for (const auto &holding_primary_replicas : dir_node->holding_primary_replicas) {
+            info.holding_primary_replica_counts[holding_primary_replicas.first] =
+                static_cast<int>(holding_primary_replicas.second.size());
+        }
+
+        for (const auto &holding_secondary_replicas : dir_node->holding_secondary_replicas) {
+            info.holding_secondary_replica_counts[holding_secondary_replicas.first] =
+                static_cast<int>(holding_secondary_replicas.second.size());
+        }
+
+        info.tag = dir_node->tag;
+        info.full_dir = dir_node->full_dir;
+        info.disk_capacity_mb = dir_node->disk_capacity_mb;
+        info.disk_available_mb = dir_node->disk_available_mb;
+
+        resp.disk_infos.emplace_back(info);
+    }
+
+    resp.total_capacity_mb = _fs_manager._total_capacity_mb;
+    resp.total_available_mb = _fs_manager._total_available_mb;
+    resp.err = ERR_OK;
```

Comment:
Is this function always got ERR_OK in response?

Reply:
So far it seems, because the code don't have other error such as `ERR_OBJECT_NOT_FOUND`. Indeed, I refer to https://github.com/XiaoMi/rdsn/blob/f6604cbda06cd5aa2ac9d64ca299ef0d98751f04/src/dist/replication/lib/replica_stub.cpp#L859

Reply:
oKay~